### PR TITLE
Update cert expiration from master

### PR DESCRIPTION
**Pull request by genexpr**
Related to https://github.com/rancher/rancher/issues/35068.

This back-ports a commit that allows to specify the webhook's certificate expiration date in days and, by default, set the certificate's expiration date to 10 years after creation. This expiration date is passed down to the dynamiclistener via the new `CATTLE_NEW_SIGNED_CERT_EXPIRATION_DAYS` environment variable.

Once the corresponding [PR](https://github.com/rancher/dynamiclistener/pull/54) for dynamiclistener is merged, I will update webhook's `go.mod` so that it requires `dynamiclistener` at the appropriate commit.

### CODE REVIEW

**Review by cmurphy — COMMENTED** [review]
Looks like another commit is needed to fix the copyright headers for `go generate` https://drone-pr.rancher.io/rancher/webhook/162/1/2

Then once go.mod is updated lgtm

**Comment by genexpr** [comment]
When this is merged, we will tag this (v0.1.3-rc1). And then update Rancher `release/v2.5` package/Dockerfile to set the `CATTLE_RANCHER_WEBHOOK_MIN_VERSION` to the tag value.

File: go.mod [hunk 1]
```diff
@@ -29,10 +29,10 @@ replace (
 
 require (
 	github.com/gorilla/mux v1.7.3
-	github.com/rancher/dynamiclistener v0.2.1-0.20201110045217-9b1b7d3132e8
+	github.com/rancher/dynamiclistener v0.2.8-0.20220107182323-a73b7d7f4c0c
```
Comment (cbron) [inline go.mod:32]: We'll need to tag this before 2.5.12 release

**Review by cbron — APPROVED** [review]

**Review by cmurphy — CHANGES_REQUESTED** [review]
Could you squash 9f9341b798a098f473334e3ceddff6a2bdc7eaab e1e7a6da370d2e2a34c417c3ffb6600e03f1aa4f and 1aed61ccc7a4e217eabe4112096ab841d6f5c800 into a single commit?

**Review by cmurphy — APPROVED** [review]

**Pull request merged**